### PR TITLE
TableCellStatus component to DRY up status indicators in table cells

### PR DESCRIPTION
**Pull request by adamabeshouse**
# Checks
- [x] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [x] Follows the [Airbnb React/JSX Style guide](https://github.com/airbnb/javascript/tree/master/react).
- [x] Make sure your commit messages end with a Signed-off-by string (this line
  can be automatically added by git if you run the `git-commit` command with
  the `-s` option)

### CODE REVIEW

**Comment by adamabeshouse** [comment]
[DRY up duplicate code for 'loading' and 'error' visualizations from table cells](https://trello.com/c/jwt0vAFo/189-dry-up-duplicate-code-for-loading-and-error-visualizations-from-table-cells)

File: src/shared/components/TableCellStatus.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+};
+
+export default class TableCellStatus extends React.Component<TableCellStatusProps, {}> {
+    public render() {
+        let alt:string;
+        let text:string;
+        switch (this.props.status) {
+            case "loading":
+                alt=this.props.loadingAlt || "Querying server for data.";
+                text="LOADING";
```
Comment (alisman) [inline src/shared/components/TableCellStatus.tsx]: Seems like people want to see a little spinner.

**Comment by alisman** [comment]
@onursumer can we use this for annotation icons?

File: src/shared/components/TableCellStatus.tsx (hunk 1 above)
Comment (adamabeshouse) [inline src/shared/components/TableCellStatus.tsx]: that would be visually very cluttered, I think.... but we can try it

Comment (onursumer) [inline src/shared/components/TableCellStatus.tsx]: We don't need to do spinner for every column. I think they specifically want to see it as an annotation icon placeholder

**Comment by onursumer** [comment]
@adamabeshouse @alisman Annotation column has more than one data source (OncoKB, Hotspots, etc.) loading in parallel. Displaying "Loading" for every icon wouldn't seem visually nice. I think we need something more compact for that column.

File: src/pages/patientView/mutation/column/DiscreteCNAColumnFormatter.tsx [hunk 2]
```diff
@@ -62,6 +63,7 @@ export default class DiscreteCNAColumnFormatter {
     }
 
     protected static getTdContents(cacheDatum:DiscreteCNACacheDataType | null) {
+        let status:"loading"|"error"|"na";
         if (cacheDatum !== null && cacheDatum.status === "complete" && cacheDatum.data !== null) {
             const alteration = cacheDatum.data.alteration;
             if (alteration === 2) {
```
Comment (alisman) [inline src/pages/patientView/mutation/column/DiscreteCNAColumnFormatter.tsx]: i think we have an enum for this somewhere

Comment (onursumer) [inline src/pages/patientView/mutation/column/DiscreteCNAColumnFormatter.tsx]: well, there is one in `CnaColumnFormatter` the and another one in `CopyNumberUtils`

Comment (alisman) [inline src/pages/patientView/mutation/column/DiscreteCNAColumnFormatter.tsx]: test/fixtures/outputs/pages/patientView/copyNumberAlterations/column/CnaColumnFormatter.tsx.html
Check with @onursumer about whether we can use this.

File: src/pages/patientView/mutation/column/CohortColumnFormatter.tsx [hunk 3]
```diff
@@ -61,31 +62,13 @@ export default class CohortColumnFormatter {
 
     private static makeCohortFrequencyViz(variantCount:AugVariantCountOutput | null) {
 
+        let status:"loading"|"error"|"na";
```
Comment (onursumer) [inline src/pages/patientView/mutation/column/CohortColumnFormatter.tsx]: shouldn't this be `let status:"loading"|"error"|"na"|undefined;`

File: src/pages/patientView/mutation/column/CohortColumnFormatter.tsx [hunk 4]
```diff
@@ -61,31 +62,13 @@ export default class CohortColumnFormatter {
...
-            return (
-                <span
-                    style={{color: "gray", fontSize:"xx-small", textAlign:"center"}}
-                    alt="mRNA data is not available for this gene."
-                >
-                    NA
-                </span>
-            );
+            status = "na";
         } else {
```
Comment (onursumer) [inline src/pages/patientView/mutation/column/CohortColumnFormatter.tsx]: can we move this else after
```JavaScript
if (status) {
   ...
}
```

File: src/pages/patientView/mutation/column/DiscreteCNAColumnFormatter.tsx (hunk 2 above)
Comment (adamabeshouse) [inline src/pages/patientView/mutation/column/DiscreteCNAColumnFormatter.tsx]: it definitely doesn't make sense to use an enum from a different component...

File: src/pages/patientView/mutation/column/CohortColumnFormatter.tsx (hunk 4 above)
Comment (adamabeshouse) [inline src/pages/patientView/mutation/column/CohortColumnFormatter.tsx]: typescript isn't smart enough to infer the type guards on variantCount, so it complains if it's not in the attached `} else {`

File: src/pages/patientView/mutation/column/CohortColumnFormatter.tsx (hunk 3 above)
Comment (adamabeshouse) [inline src/pages/patientView/mutation/column/CohortColumnFormatter.tsx]: I'm adding an enum

**Comment by adamabeshouse** [comment]
@onursumer @alisman changes made

**Pull request merged**